proxy/src/main/java/org/apache/rocketmq/proxy/processor/ProducerProcessor.java

```diff
@@ -102,11 +108,18 @@ public CompletableFuture<List<SendResult>> sendMessage(ProxyContext ctx, QueueSe
                         if (SendStatus.SEND_OK.equals(sendResult.getSendStatus()) &&
                             tranType == MessageSysFlag.TRANSACTION_PREPARED_TYPE &&
                             StringUtils.isNotBlank(sendResult.getTransactionId())) {
-                            fillTransactionData(ctx, producerGroup, messageQueue, sendResult, messageList);
+                            fillTransactionData(ctx, producerGroup, finalMessageQueue, sendResult, messageList);
                         }
                     }
+                    endTimestamp.set(System.currentTimeMillis());
+                    this.serviceManager.getTopicRouteService().updateFaultItem(finalMessageQueue.getBrokerName(),endTimestamp.get() - beginTimestampFirst, false, true);
                     return sendResultList;
-                }, this.executor);
+                }, this.executor)
+                    .exceptionally(t -> {
```

Comment:
Use whenComplete is better. When an exception occurs, the upper layer function will get null in thApply